dist/amplify.js

```diff
@@ -0,0 +1,110 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.createAmplifyApp = createAmplifyApp;
+const aws = require("@pulumi/aws");
+const cloudflare = require("@pulumi/cloudflare");
+const std = require("@pulumi/std");
+function createAmplifyApp(config) {
+    const name = `${config.prefix}.${config.domainName}`;
+    const buildSpec = `version: 1
+applications:
+  - frontend:
+      phases:
+        preBuild:
+          commands:
+            - ${config.preBuildCommand || ''}
+        build:
+          commands:
+            - ${config.buildCommand}
+      artifacts:
+        baseDirectory: ${config.artifactsBaseDirectory}
+        files:
+          - '**/*'
+      cache:
+        paths:
+          - node_modules/**/*
+    appRoot: ${config.monorepoAppRoot || 'app'}
+`;
+    const amplifyApp = new aws.amplify.App(name, {
+        name: name,
+        repository: config.githubUrl,
+        oauthToken: config.githubAccessToken,
+        platform: config.platform || "WEB_COMPUTE",
+        buildSpec: buildSpec,
+        customRules: config.customRules || [
+            {
+                source: "/<*>",
+                status: "404",
+                target: "/index.html",
+            },
+        ],
+        environmentVariables: Object.assign({ AMPLIFY_DIFF_DEPLOY: "false", AMPLIFY_MONOREPO_APP_ROOT: config.monorepoAppRoot || "app" }, config.environmentVariables),
+        enableBasicAuth: config.enableBasicAuth || false,
```

Comment:
I think we can remove this for now. We have no use-case for basic auth & will drop 3 parameter (easier to configure it ). 
But it's ok for me if we leave it as it is now